.github/workflows/nightly-benchmark-cleanup.yml

```diff
@@ -0,0 +1,168 @@
+        id: logs
+        run: |
+          # Wait for job logs with retries
+          max_attempts=12
+          attempt=1
+          while [ $attempt -le $max_attempts ]; do
+            if kubectl logs ${{ inputs.nodeId }} --namespace cloud-benchmark > benchmark.log 2>/dev/null; then
+              if grep -q '"time-taken-ms":' benchmark.log; then
+                time_taken=$(grep -o '"time-taken-ms":[0-9]\+' benchmark.log | tail -1 | cut -d: -f2)
+                echo "time_taken=$time_taken" >> "$GITHUB_OUTPUT"
+                break
+              fi
+            fi
+            echo "Attempt $attempt: Waiting for benchmark logs..."
+            sleep 5
+            attempt=$((attempt + 1))
+          done
+
+      - name: Set Timestamp
+        id: timestamp
+        run: |
+          echo "value=$(date -u +"%Y-%m-%d-%H-%M-%S")" >> "$GITHUB_OUTPUT"
+          echo "date=$(date -u +"%Y/%m/%d")" >> "$GITHUB_OUTPUT"
+
+      - name: Upload Benchmark Logs
+        id: upload
+        if: ${{ always() && steps.check-deployment.outcome == 'success' }}
+        continue-on-error: true
+        uses: azure/CLI@v1
+        with:
+          inlineScript: |
+            set -euo pipefail
+            test -f benchmark.log || { echo "benchmark.log not found"; exit 1; }
+            az storage blob upload \
+              --connection-string "${{ secrets.BENCHMARK_AZURE_STORAGE_CONNECTION_STRING }}" \
+              --container-name benchmark-logs \
+              --file benchmark.log \
+              --name "${{ inputs.benchType }}/${{ steps.timestamp.outputs.date }}/${{ steps.timestamp.outputs.value }}-${{ inputs.nodeId }}-${{ inputs.status }}.log" \
+              --content-type "text/plain; charset=utf-8" \
+              --overwrite true
+
+      - name: Generate read-only SAS URL for the log
+        id: sas
+        if: ${{ always() && steps.check-deployment.outcome == 'success' }}
+        uses: azure/CLI@v1
+        with:
+          inlineScript: |
+            set -euo pipefail
+            CONN='${{ secrets.BENCHMARK_AZURE_STORAGE_CONNECTION_STRING }}'
+            CONTAINER=benchmark-logs
+            NAME="${{ inputs.benchType }}/${{ steps.timestamp.outputs.date }}/${{ steps.timestamp.outputs.value }}-${{ inputs.nodeId }}-${{ inputs.status }}.log"
+            EXP=$(date -u -d '+7 days' '+%Y-%m-%dT%H:%MZ')
+            URL=$(az storage blob generate-sas \
+                  --connection-string "$CONN" \
+                  --container-name "$CONTAINER" \
+                  --name "$NAME" \
+                  --permissions r \
+                  --expiry "$EXP" \
+                  --https-only \
+                  --full-uri \
+                  -o tsv)
+            echo "url=$URL" >> "$GITHUB_OUTPUT"
+
+      - name: Run Cleanup Script
+        if: steps.check-deployment.outcome == 'success'
+        run: |
+          # Since we're using Azure for benchmarks
+          ./modules/bench/cloud/clear-bench.sh azure
```

Comment:
If the rest of this workflow fails - are we left with a hanging helm deployment that doesn't get cleared up?

Reply:
I'll fix this!!